services/venice-controller/src/main/java/com/linkedin/venice/controller/server/StoresRoutes.java

```diff
@@ -431,6 +431,15 @@ public void internalHandle(Request request, StoreMigrationResponse veniceRespons
           veniceResponse.setErrorType(ErrorType.BAD_REQUEST);
           return;
         }
+        // Both source and destination clusters should either have RT versioning enabled or disabled
+        if (admin.getControllerConfig(srcClusterName).isRealTimeTopicVersioningEnabled() != admin
+            .getControllerConfig(destClusterName)
```

Comment:
We will want to refactor the controller configs framework in future, but detailed design is not there yet. Here, we have an assumption that the leader controller of src cluster will also have access to configs of destination cluster, which is true now.
Could you please add a check to see if `admin.getControllerConfig(destClusterName)` returns null? If so, fail open and let the store migration to go through?